examples/satellites/build_satellites_bdb.py

```diff
@@ -120,24 +162,21 @@ def execute(bql):
 
 execute('INITIALIZE %d MODELS FOR satellites_cc' % (num_models,))
 
-execute('ANALYZE satellites_cc FOR %d ITERATIONS CHECKPOINT 5 ITERATIONS WAIT'
-        % num_iters)
-
-# create a diagnostics plot
-plot_file_name = out_file_name('satellites', '-logscores.pdf')
-log('writing diagnostic plot to %s' % plot_file_name)
-fig = bdbcontrib.plot_crosscat_chain_diagnostics(bdb, 'logscore',
-    'satellites_cc')
-plt.savefig(plot_file_name)
-
-log('closing bdb %s' % bdb_file)
-bdb.close()
-
-metadata_file = out_file_name('satellites', '-meta.txt')
-sha_sum = subprocess.check_output(["sha256sum", bdb_file])
-total_time = time.time() - then
-def record_metadata(f):
-    f.write("DB file " + bdb_file + "\n")
+cur_iter_ct = 0
+
+def snapshot():
+    log('vacuuming')
+    bdb.sql_execute('vacuum')
+    cur_infix = '-%dm-%di' % (num_models, cur_iter_ct)
+    save_file_name = out_file_name('satellites', cur_infix + '.bdb')
+    meta_file_name = out_file_name('satellites', cur_infix + '-meta.txt')
+    log('recording snapshot ' + save_file_name)
+    os.system("cp %s %s" % (bdb_file, save_file_name))
+    report(save_file_name, meta_file_name)
+
+def record_metadata(f, saved_file_name, sha_sum, total_time,
+                    plot_file_name=None):
+    f.write("DB file " + saved_file_name + "\n")
```

Comment:
History: First this was just what it wrote to stdout on completion, then I thought "hey, let's save it".  You're right that json would be better.